### PR TITLE
chore: migrate tslint to eslint

### PR DESCRIPTION
**Pull request by the-ult**
### What

Migrate from TSLint to ESLint
_and upgrade dependencies_

### Why

To provide the 'best' examples possible, we should make sure the examples are making use of the best practices. Like discussed [here](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-library-eslint-plugins)
To help doing that, there are 2 great ESLint plugins:
- [eslint-plugin-testing-library](https://github.com/testing-library/eslint-plugin-testing-library) 
- [eslint-plugin-jest-dom](https://github.com/testing-library/eslint-plugin-jest-dom)

To be able to use these plugins we first need to migrate from TSLint to ESLint.

This PR is a prerequisite of #198 (Adding the above plugins and improve the examples)

### How

- Upgrade to NX 12.0.3
   - _Needed to be able to use the `@nrwl/angular:convert-tslint-to-eslint` schematic_ 
- Move  `apps/example-app/app/examples` to `apps/example-app/app/examples`
   - _Needed to be able to use the `@nrwl/angular:convert-tslint-to-eslint` schematic since the schematic expects the 'NX' structure for an app_
- Migrate TSLint to ESLint
  - example-app
  - jest-utils
  - testing-library
- ESLint
  - Remove duplicate rules which are already provided by _NX_ or other [plugin]/recommended    

### Troubleshooting

*Removing TSLint*
Could not remove `tslint` from the `package.json` due to the following errors in a migration file:
```
projects/testing-library/migrations/4_0_0/rules/noComponentParametersRule.ts
(2,49): error TS2307: Cannot find module 'tslint' or its corresponding type declarations.
projects/testing-library/migrations/4_0_0/rules/noComponentParametersRule.ts
(38,82): error TS2339: Property 'ruleName' does not exist on type 'Rule'.
projects/testing-library/migrations/4_0_0/rules/noComponentParametersRule.ts
(39,66): error TS2339: Property 'ruleName' does not exist on type 'Rule'.
projects/testing-library/migrations/4_0_0/rules/noComponentPropertyRule.ts
(2,49): error TS2307: Cannot find module 'tslint' or its corresponding type declarations.
projects/testing-library/migrations/4_0_0/rules/noComponentPropertyRule.ts
(19,69): error TS2339: Property 'ruleName' does not exist on type 'Rule'.
projects/testing-library/migrations/4_0_0/rules/noCreateComponentRule.ts
(2,49): error TS2307: Cannot find module 'tslint' or its corresponding type declarations.
projects/testing-library/migrations/4_0_0/rules/noCreateComponentRule.ts
(27,69): error TS2339: Property 'ruleName' does not exist on type 'Rule'.
projects/testing-library/migrations/4_0_0/rules/noCreateComponentRule.ts
(37,69): error TS2339: Property 'ruleName' does not exist on type 'Rule'.
projects/testing-library/migrations/5_1_2/rules/noAngularExtensionsImportRule.ts
(2,49): error TS2307: Cannot find module 'tslint' or its corresponding type declarations.
projects/testing-library/migrations/5_1_2/rules/noAngularExtensionsImportRule.ts
(23,69): error TS2339: Property 'ruleName' does not exist on type 'Rule'.
```

### Checklist
- [x] Tests 
- [x] Check / improve the ESLint rules
   - Are there any more duplicates which can be removed? 
- [x] Documentation
   - [ ] Adjust documentation on [testing-library](https://testing-library.com/docs/angular-testing-library/examples) to the [github examples](https://github.com/testing-library/angular-testing-library/tree/master/apps/example-app/app/examples)
      - `src` is added as root of example-app -> `apps/example-app/app/examples/src` instead of `apps/example-app/app/examples`   
 - [x] Ready to merge

### CODE REVIEW

File: apps/example-app/src/app/examples/01-nested-component.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Component, Input, Output, EventEmitter } from '@angular/core';
 
 @Component({
   selector: 'app-button',
-  template: ` <button (click)="raise.emit()">{{ name }}</button> `,
+  template: ' <button (click)="raise.emit()">{{ name }}</button> ',
```
Comment (the-ult) [inline apps/example-app/src/app/examples/01-nested-component.ts:5]: The replacement of the back-tick ``` was automatically done by `lint --fix`

File: projects/testing-library/src/lib/models.ts [hunk 2]
```diff
@@ -250,6 +250,7 @@ export interface RenderComponentOptions<ComponentType, Q extends Queries = typeo
   removeAngularAttributes?: boolean;
 }
 
+// eslint-disable-next-line @typescript-eslint/ban-types
```
Comment (the-ult) [inline projects/testing-library/src/lib/models.ts:253]: Added due to the ESLint error about the usage of `object`

**Review by timdeschryver — COMMENTED** [review]
I noticed that the "ready to merge" checkbox isn't checked.
What would still need to be done in order to get this merged?

From my point of view, this is looking great already and I would lean towards removing TSLint and the migrations.

**Comment by the-ult** [comment]
Wasn't sure I could check it or the reviewer (you) should 😇 
I'll try and remove the migration and the TSLint and fix the README

**Comment by the-ult** [comment]
BTW:
I tried updating the:
```console
@testing-library/dom          7.29.4  7.29.4  7.30.3  node_modules/@testing-library/dom         angular-testing-library
@testing-library/user-event   12.8.3  12.8.3  13.1.2  node_modules/@testing-library/user-event  angular-testing-library
```

but these updates caused issues with `fakeAsync` etc. So I thought it would be best to leave these updates to separate PR by someone with more knowledge of the `testing-library` 😇

**Comment by the-ult** [comment]
Should we/somebody update the documentation on the [testing-library/docs page](https://testing-library.com/docs/angular-testing-library/examples) after merging?

**Comment by timdeschryver** [comment]
Yep, that' on the radar to take a look why those tests are failing with the latest version of dom testing library (that's why it's currently pinned). 
I'll give this another look tomorrow morning, but I think this can be merged 🎉
Thanks in advance!

Once it's merged, we'll update the docs.

**Comment by the-ult** [comment]
After it is merged I'll start the PR with the testing-library eslint rules 👍

**Review by timdeschryver — APPROVED** [review]

**Comment by timdeschryver** [comment]
Thanks @the-ult !
@all-contributors please add @the-ult  for code

**Comment by the-ult** [comment]
@timdeschryver I adjusted the url on [testing-library.com](https://github.com/testing-library/testing-library-docs/pull/795) as well.

**Pull request merged**